doomed_patient/doomed_patient/Player.cpp

```diff
@@ -5,11 +5,21 @@
 
 Player::Player(PatientGame* game, VectorXY startCoordinates)
 	: Character(game, startCoordinates, game->getPlayerSprite()),
-	speed(2)
+	speed(0)
 {
+	currentCell = game->level.getCell(VectorXY(0, 0));
+	centre = currentCell->getCentre();
 }
 
 void Player::move(Directions::Direction direction)
 {
-	centre += Directions::getDirectionVector(direction) * speed;
+	
+	// Put collision detection here
+
+	if (currentCell->getEdge(direction)->isWall())
+	{
+		return;
+	}
+	currentCell = game->level.getCell(currentCell->getCoordinates() + Directions::getDirectionVector(direction));
+	centre = currentCell->getCentre();
```

Comment:
Checking whether the cell has walls is good if the player has restricted movement (like in my Kivy game), or for NPCs where the system is fully in control of the movement. But if the player can move freely, it's probably better to check collisions using the space the sprite takes up... 

So the question is, which route would you prefer to go down:

**option 1:** restrict the characters movement to up, left, down and right and only allow them to move in intervals of one cell at a time (i.e. they can't stop in between cells), and then do collision checking based on the cell's edges

**option 2:** allow the player to move freely, and do collision checking using the rectangle the sprite takes up vs the rectangle the object takes up

In my opinion, option 2 will be easier and quicker to implement. Plus, it gives the player an advantage over the NPCs.